o4/o4.py

```diff
@@ -634,6 +649,14 @@ class LogAndAbort(Exception):
                     # its objects before operation, as in "And for my next act... !"
                     # This premature printing leads to false checksum errors during sync.
                     print(fstat_join(fstat))
+    # TODO: This should probably be in a finally clause
+    for fname, omode in orig_mode.items():
```

Comment:
This is removing write access from files that didn't get synced?

Reply:
Yes, I'm being a chicken.